Backend/TemporalDataManager.py

```diff
@@ -0,0 +1,115 @@
+                snapid += 1
+                time.sleep(24)   # sleep for 24 seconds to avoid Twitter get/trends rate limit
+
+    # TODO: periodically pull tweets at different dates
+    # TODO: this method template is an oversight and it might not be necessary
+    def periodic_tweets_retrieval(self):
+        ()
+
+    #       get the timeline of the trend. Return all occurrences of the trend given a datetime span
+    #       trend: the trend keyword
+    #       TODO: woeid: the location of the trend (optional) NOT YET SUPPORTED AND MIGHT NOT BE NECESSARY
+    #       fromdate: the datetime to search from
+    #       todate: the datetime to search to
+    #       days, hours, minutes, seconds = the time step of each unit
+    #       return value: return a csv format string in format:  datetime,val1,val2,val3
+    def get_trends_snapshot_as_csv(self, trends, fromdate, todate, woeid=0, days=0, hours=2, minutes=0, seconds=0):
+        # call algo manager to query to database and return trends within the time frame
+        # snapres return an array of database request for trend_snapshot data
+        # each element are for each trend in the trends argument
+        snapsres = self.algo.get_trends_snapshot_from_database(trends, fromdate, todate, woeid)
+
+        # preload the precv dictionary
+        precsv = {}             # dictionary to be parsed into csv. contain date and array of tweet volume values
+        maxvals = len(trends)   # max amount of value for each dictionary entry
+        tempdate = fromdate
+        while tempdate < todate:
+            precsv[tempdate] = []
+            tempdate += timedelta(days=days, hours=hours, minutes=minutes, seconds=seconds)
+        for csventry in precsv:
+            for i in range(maxvals):
+                precsv[csventry].append(0)
+
+        i = 0
+        for key in snapsres.keys():
+            rows = snapsres[key]
+            lenrows = len(rows)
+
+            print("Snapshots retrieved for {}: {}".format(trends[i], lenrows))
+
+            snapsbucket = self.algo.get_snapstime_bucket_from_database_tuples(rows, fromdate, days=days, hours=hours, minutes=minutes, seconds=seconds)
+            prevval = 0
+            for date in snapsbucket.keys():
+                snaps = snapsbucket[date]
+                totalvol = 0
+                for snap in snaps:
+                    totalvol += snap[4]
+                precsv[date][i] = totalvol    # updating the tweet volume value normally with just totalvol
+
+                # code comment below is for saving previous totalvol value.
+                # if new value is 0 then reuse the previous totalvol value
+                """
+                if totalvol == 0:
+                    precsv[date][i] = prevval
+                else:
+                    precsv[date][i] = totalvol
+                    prevval = totalvol
+                """
+            i += 1  # increment the trend index
+            print()
```

Comment:
what is this for?

Reply:
this is for the csv file. each row lead with a datetime, and arbitrary values after.
this is just an indexer for those values

Reply:
I meant the empty print() call

Reply:
oh its just for new line to make the console output nicer